http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java

```diff
@@ -287,4 +298,39 @@ private static HttpTransport defaultHttpClientFromConfig(LightblueClientConfigur
         }
     }
 
+    private Execution getExecutionInstance() {
+        if (execution == null) {
+            synchronized (this) {
+                if (execution == null) {
+                    MongoExecution mongoExecution = null;
+                    if (configuration.getReadPreference() != null) {
+                        mongoExecution = MongoExecution.withReadPreference(configuration.getReadPreference());
+                    }
+
+                    if (configuration.getWriteConcern() != null) {
+                        if(mongoExecution == null){
+                            mongoExecution = MongoExecution.withWriteConcern(configuration.getWriteConcern());
+                        }
+                        else {
+                            mongoExecution.addWriteConcern(configuration.getWriteConcern());
+                        }
+                    }
+
+                    if (configuration.getMaxQueryTimeMS() != null) {
+                        if(mongoExecution == null){
+                            mongoExecution = MongoExecution.withMaxQueryTimeMS(configuration.getMaxQueryTimeMS());
+                        }
+                        else {
+                            mongoExecution.addMaxQueryTimeMS(configuration.getMaxQueryTimeMS());
+                        }
+                    }
+
+                    execution = mongoExecution;
+                }
+            }
+        }
+
+        return execution;
+    }
```

Comment:
This is unnecessarily complex. Let's simplify by allowing empty Execution creation. If you use package visibility, the constructor will be protected from misuse outside the client.